tasks/src/main/scala/com/itv/bucky/tasks/Task.scala

```diff
@@ -0,0 +1,98 @@
+package com.itv.bucky.tasks
+
+import com.itv.bucky.PayloadMarshaller.StringPayloadMarshaller
+import com.itv.bucky.PublishCommandBuilder.publishCommandBuilder
+import com.itv.bucky._
+import com.itv.bucky.decl._
+import com.itv.bucky.pattern.requeue.{RequeuePolicy, _}
+import com.typesafe.scalalogging.StrictLogging
+
+import scala.concurrent.duration._
+import scala.language.higherKinds
+
+final case class TaskName(value: String)
+
+class Task[T](
+  name: TaskName,
+  exchangeName: Option[ExchangeName] = None,
+  routingKey: Option[RoutingKey] = None,
+  queueName: Option[QueueName] = None,
+  exchangeType: Option[ExchangeType] = None,
+  requeuePolicy: Option[RequeuePolicy] = None,
+  prefetchCount: Option[Int] = None
+)(implicit
+  val marshaller: PayloadMarshaller[T],
+  val unmarshaller: PayloadUnmarshaller[T]
+)
+    extends StrictLogging {
+
+  def getExchangeName: ExchangeName =
+    exchangeName.getOrElse(ExchangeName(s"bucky.tasks.exchange.${name.value}"))
+  def getRoutingKey: RoutingKey =
+    routingKey.getOrElse(RoutingKey(s"bucky.tasks.route.${name.value}"))
+  def getQueueName: QueueName =
+    queueName.getOrElse(QueueName(s"bucky.tasks.queue.${name.value}"))
```

Comment:
Have mixed opinions over whether it's a good idea to have default values for these.. thoughts? probably being fussy now

Reply:
I'm a big fan of convention over configuration

Reply:
:+1: